catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStoreImpl.java

```diff
@@ -136,6 +161,31 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
             return new SourceResponseImpl(request, Collections.emptyList());
         }
 
+        SourceResponse registryQueryResponse = queryResponse(request);
+        for (Result singleResult : registryQueryResponse.getResults()) {
+            if (singleResult.getMetacard()
+                    .getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                    .getValue()
+                    .toString()
+                    .equals(registryId)) {
```

Comment:
Just one. I could add a break so it will stop as soon as it finds it, but I don't know if/how that would be better than a stream solution. Either way, I will update. 

Also there is an open ticket to remove this code altogether and use an OSGI EventHandler to listen for changes to the remoteName.